### ISSUE TITLE
Add kind_detail values for pois & other feature types

### ISSUE THREAD

**Issue by nvkelso** [tilezen/vector-datasource#719]
**tl;dr**

Add new `kind_detail` for:

-  `restaurants` should additionally map the `cuisine` values into `kind_detail`
- `pitches` should additionally map the `sport` values into `kind_detail`

===

We currently provide more granular details needed for displaying roads in various `highway`, `aerialway`, & etc tags. Similarly `restaurants` can be paired with a `cuisine` value for greater granularity. For pitches, the `sport` value. Let's formalize that and store in a generic `kind_detail` property that can be referenced. `kind_detail` values should be unique (within a parent `kind`, and across parent `kind`s).

This can be done for the `pois` layer now, `roads` layer soon. Other layers and victory!

_(An earlier version of this referred to `subkind` instead of `kind_detail`.)_

**Comment by nvkelso** [comment]
Related work was done in https://github.com/tilezen/vector-datasource/issues/970 for roads.

**Comment by iandees** [comment]
Should the new key be `subkind` or `kind_detail`?

**Comment by nvkelso** [comment]
@iandees We've standardized on `kind_detail`.

**Comment by iandees** [comment]
👍  I updated the description to reference `kind_detail` rather than `subkind` so I don't confuse my future self.

**Comment by nvkelso** [comment]
Verified :)
